ui/ui-frontend/projects/referential/src/app/agency/agency.component.html

```diff
@@ -28,15 +28,15 @@ <h5>{{ 'AGENCY.HOME.TITLE' | translate }}</h5>
           <button mat-menu-item [disabled]="!hasImportRole" (click)="openAgencyImportDialog()">
             {{ 'AGENCY.HOME.ACTION_IMPORT' | translate }}
           </button>
-          <button mat-menu-item [disabled]="!hasExportRole" (click)="exportAgencies()">
+          <button mat-menu-item [disabled]="noResult || !hasExportRole" (click)="exportAgencies()">
             {{ 'AGENCY.HOME.ACTION_EXPORT' | translate }}
           </button>
         </vitamui-menu-button>
       </vitamui-banner>
     </div>
 
     <div class="vitamui-content">
-      <app-agency-list (agencyClick)="showAgency($event)" [search]="search"></app-agency-list>
+      <app-agency-list (agencyClick)="showAgency($event)" (noResult)="onDataSourceChange($event)" [search]="search"></app-agency-list>
```

Comment:
C'est beaucoup trop compliqué. On peut supprimer tout le code modifié dans les autres fichiers et remplacer par :
```suggestion
          <button mat-menu-item [disabled]="!agencyList?.dataSource?.length || !hasExportRole" (click)="exportAgencies()">
            {{ 'AGENCY.HOME.ACTION_EXPORT' | translate }}
          </button>
        </vitamui-menu-button>
      </vitamui-banner>
    </div>

    <div class="vitamui-content">
      <app-agency-list #agencyList (agencyClick)="showAgency($event)" [search]="search"></app-agency-list>
```